active/0000-building-target-assets.md

```diff
@@ -0,0 +1,187 @@
+
+If an addon wants to define a new target, it will specify the new target in its package.json (under `ember-addon` config). The definition would look as follows in the package.json:
+```
+"ember-addon": {
+  "configPath": "tests/dummy/config",
+  "additionalTargets": {
+    "fastboot": {
+      "dependsOn": ["browser"]
+    }
+  }
+}
+```
+The above config means that this ember-addon wants to define an additional target named "fastboot" which depends on the "browser" target. The depends on means that whenever this target is built, it needs to make sure the dependent target is also built so that the assets can be served correctly in the target environment.
+
+If an app wants to define a new target, it will specify the new target in `ember-cli-build.js` when it creates the ember app:
+```
+var app = new EmberApp(defaults, {
+  minifyJS: {
+    enabled: false
+  },
+  minifyCSS: {
+    enabled: false
+  },
+  additionalTargets: {
+    "fastboot": {
+      "dependsOn": ["browser"]
+    }
+  }
+});
+
+//...
+return app.toTree();
+```
+
+## Additional options to existing CLI commands
+The existing ember-cli commands only assume there is one target that it needs to build/generate files for. With the concept of targets, we now want to now allow the CLI commands to allow targets to be generated and created. The following will be the workflow for a ember user intending to create and use target.
+
+1. A user can generate a target using the following command: `ember generate target <target name>`
+This will end up creating a `fastboot` directory with app folder inside if it is done in an app or app and addon directories if done in an addon. Note: The target name should be the same as the one registered by the app/addon.
+
+2. A user can generate a route in a target environment: `ember generate route <route name> --target <target name>`
+This generates a route in the targeted environment.
+
+3. Building a target: `ember build --target <target name>`
+This will build the target artifacts. If the target depends on other targets, those will also be built as part of this command. The generated assets will all be under `dist/assets` with the following file naming:
+```
+dist/
+|-- assets/
+│   |-- vendor.js
+│   |-- app.js
+│   |-- vendor-<targetname>.js
+│   |-- app-<targetname>.js
+```
+
+Note:
+1. The command `ember serve` will remain as it is today since it serves the assets based on what is requested. If the targeted assets need to be served, the app/addon should update the index.html or their environment in which this is loaded to be served. The targeted app and addon directories should be watched so that a change in them triggers a rebuild.
```

Comment:
I don't see how this cleanly works. The `index.html` js and css loading are static to two statements each. It doesn't have a way to dynamically load the additional target files. If your `generate target` command added lines to this file, then it is now permanently configured for the target only, and running in browser mode would try to request the target files and 404.